source/opt/aggressive_dead_code_elim_pass.cpp

```diff
@@ -146,8 +146,9 @@ void AggressiveDCEPass::ComputeBlock2HeaderMaps(
       currentMergeInst.push(mergeInst);
       currentBranchInst.push(branchInst);
     }
-    block2headerMerge_[*bi] = currentMergeInst.top();
-    block2headerBranch_[*bi] = currentBranchInst.top();
+    ir::Instruction* cbi = currentBranchInst.top();
+    block2headerBranch_[*bi] = cbi;
+    branch2merge_[cbi] = currentMergeInst.top();
```

Comment:
I'm lost at this point.  Some comments will probably be needed.  Why is this line executed for every block.  Won't it be setting the same entry to the same value multiple times?

Also, now that I look at it closer, why do we need three different stack on the go.  The current branch instruction will always be the instruction after the merge instruction.  The merge block is an operand to the merge instruction.  Can't we just keep one of them?

Then at the same time, why have branch2merge_ at all?  Can't we just step back one instruction from the branch instruction?

Reply:
> Can't we just step back one instruction from the branch instruction?

Does that work? If I have Instruction* ip pointing at an instruction, can I do --ip and have it point at the preceding instruction?

Reply:
Re: multiple updates: Yeah, when I changed to branch2merge_, I spaced that it didn't need to get updated for every block.

I got rid of two of the stacks. I would like to leave the branch2merge_ map because it saves me some code in a couple other places in the class.

Reply:
You cannot do `--inst` with an instruction pointer, but you can do `inst->NextNode()` and `inst->PreviousNode()` to the get next and previous instruction in the basic block.  They do not cross basic block boundaries.